### ISSUE TITLE
Error while training resulting in nans as output

### ISSUE THREAD

**Issue by FoppeCrezee** [arturomoncadatorres/deepsurvk#83]
#### Describe the bug

While training over multiple epochs, the training process stops and resulting in a loss equal to nan and a model which only has nan as output. 

#### Steps/code to reproduce

Using the examples/01_using_deepsurvk.py files, you run into this error in line 153. I also encountered it while running my own model. Specifying hyperparameters like in the second notebook seems to help a bit with the problem. The error seems to be random atm, sometimes occuring after the 20th batch and sometimes completing all 1000.

```
X_train, Y_train, E_train, = load_whas(partition='training')
X_test, Y_test, E_test = load_whas(partition='testing')

# Calculate important parameters.
n_patients_train = X_train.shape[0]
n_features = X_train.shape[1]

# Standardization
X_scaler = StandardScaler().fit(X_train)
X_train = X_scaler.transform(X_train)
X_test = X_scaler.transform(X_test)

Y_scaler = StandardScaler().fit(Y_train.values.reshape(-1, 1))
Y_train = Y_scaler.transform(Y_train)
Y_test = Y_scaler.transform(Y_test)

Y_train = Y_train.flatten()
Y_test = Y_test.flatten()

# Added:
E_scaler = StandardScaler().fit(E_train.values.reshape(-1, 1))
E_train = E_scaler.transform(E_train)
E_test = E_scaler.transform(E_test)
E_train = E_train.flatten()
E_test = E_test.flatten()

sort_idx = np.argsort(Y_train)[::-1]
X_train = X_train[sort_idx]
Y_train = Y_train[sort_idx]
E_train = E_train[sort_idx]

dsk = deepsurvk.DeepSurvK(n_features=n_features, E=E_train)

callbacks = deepsurvk.common_callbacks()

epochs = 1000
history = dsk.fit(X_train, Y_train, 
                  batch_size=n_patients_train,
                  epochs=epochs, 
                  callbacks=callbacks,
                  shuffle=False)
```

#### Expected results
A trained model

#### Actual results
```
Epoch 19/1000
1/1 [==============================] - ETA: 0s - loss: -15631050.000 - 0s 47ms/step - loss: -15631050.0000
Epoch 20/1000
1/1 [==============================] - ETA: 0s - loss: -19184898.000 - 0s 42ms/step - loss: -19184898.0000
Epoch 21/1000
Batch 0: Invalid loss, terminating training
1/1 [==============================] - ETA: 0s - loss: nan
C:\Users\fcr2102.55146\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\keras\callbacks.py:1291: RuntimeWarning: invalid value encountered in less
  if self.monitor_op(current, self.best):
1/1 [==============================] - 0s 5ms/step - loss: nan
```

#### Versions
* DeepSurvK version: 0.2.0
* Python version: 3.7
* Operating System: Windows 10 Enterprise
* sklearn: 0.23.2

**Comment by arturomoncadatorres** [comment]
I haven't been able to replicate this issue consistently. However, this has also [been reported in the original DeepSurv](https://github.com/jaredleekatzman/DeepSurv/issues/14).

Apparently, a potentially good solution for this is to *not*  standardize your data during the pre-procesing, but rather normalizing it (i.e., make sure that features are in the range 0-1). However, remember that scaling is particularly sensitive to outliers, so be careful!

This is now mentioned in the example notebook [`01_deepsurvk_quickstart`](https://github.com/arturomoncadatorres/deepsurvk/blob/master/examples/01_deepsurvk_quickstart.ipynb) (commit a5d966d8fbcc8933fde01c4b9909cc1a7b2bf74f)

If you or anyone has another potential solution/alternative, I am happy to reopen this issue.
